homeassistant/components/vlc_telnet/media_player.py

```diff
@@ -0,0 +1,231 @@
+              | SUPPORT_CLEAR_PLAYLIST | SUPPORT_PLAY \
+              | SUPPORT_SHUFFLE_SET
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default='127.0.0.1'): cv.string,
+    vol.Optional(CONF_PORT, default='4212'): cv.positive_int,
+    vol.Required(CONF_PASSWORD, default='test'): cv.string,
+    vol.Optional(CONF_NAME, default='VLC-TELNET'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the vlc platform."""
+    add_entities([VlcDevice(config.get(CONF_NAME),
+                            config.get(CONF_HOST),
+                            config.get(CONF_PORT),
+                            config.get(CONF_PASSWORD))])
+
+
+class VlcDevice(MediaPlayerDevice):
+    """Representation of a vlc player."""
+
+    def __init__(self, name, host, port, passwd):
+        """Initialize the vlc device."""
+        self._instance = None
+        self._name = name
+        self._volume = None
+        self._muted = None
+        self._state = STATE_UNAVAILABLE
+        self._media_position_updated_at = None
+        self._media_position = None
+        self._media_duration = None
+        self._host = host
+        self._port = port
+        self._password = passwd
+        self._vlc = None
+        self._volume_bkp = 0
+        self._media_artist = ""
+        self._media_title = ""
+        print("loaded!")
+
+    def update(self):
+        """Get the latest details from the device."""
+        from python_telnet_vlc import VLCTelnet, ConnectionError as ConnErr
+
+        if self._vlc is None:
+            try:
+                self._vlc = VLCTelnet(self._host, self._password, self._port)
+                self._state = STATE_IDLE
+            except ConnErr:
+                self._state = STATE_UNAVAILABLE
+        else:
+            try:
+                status = self._vlc.status()
+                if status:
+                    if 'volume' in status:
+                        self._volume = int(status['volume']) / 500.0
+                    else:
+                        self._volume = None
+                    if 'state' in status:
+                        state = status["state"]
+                        if state == "playing":
+                            self._state = STATE_PLAYING
+                        elif state == "paused":
+                            self._state = STATE_PAUSED
+                        else:
+                            self._state = STATE_IDLE
+                    else:
+                        self._state = STATE_IDLE
+
+                self._media_duration = self._vlc.get_length()
+                self._media_position = self._vlc.get_time()
+
+                info = self._vlc.info()
+                if info is not None and info:
+                    if 'artist' in info[0]:
+                        self._media_artist = info[0]['artist']
+                    else:
+                        self._media_artist = None
```

Comment:
Can be simplified:
```suggestion
                    self._media_artist = info[0].get('artist')
```

Same below

Reply:
Simplified both similar calls!